internal/rpc/rpcserver/server.go

```diff
@@ -1599,21 +1607,64 @@ func (s *walletServer) PurchaseTickets(ctx context.Context,
 		lock <- time.Time{} // send matters, not the value
 	}()
 	err = s.wallet.Unlock(ctx, req.Passphrase, lock)
-	if err != nil {
-		return nil, translateError(err)
+	dontSignTx := req.DontSignTx
+
+	request := &wallet.PurchaseTicketsRequest{
+		Count:         numTickets,
+		SourceAccount: req.Account,
+		VotingAddress: ticketAddr,
+		MinConf:       minConf,
+		Expiry:        expiry,
+		DontSignTx:    dontSignTx,
+		VSPAddress:    poolAddr,
+		VSPFees:       poolFees,
+	}
+
+	if !dontSignTx {
+		lock := make(chan time.Time, 1)
+		defer func() {
+			lock <- time.Time{} // send matters, not the value
+		}()
+		err = s.wallet.Unlock(ctx, req.Passphrase, lock)
```

Comment:
The wallet is still being unlocked earlier in this method, even when signing is disabled. I think you forgot to remove that part.

Reply:
Yeah, I did. Thanks for noticing.